### ISSUE TITLE
.Net8 in-proc hosts creates an Activity without a source name

### ISSUE THREAD

**Issue by JessicaA-Auror** [Azure/azure-functions-host#10571]
When upgrading an http triggered in-proc function to net8, the host runtime is now creating a new `Activity` without a `Source Name`. This is a change in behaviour that breaks telemetry. This changes otel by adding a new parent activity. Additionally the activity cannot be subscribe to due to not having a source name. When exporting otel, this looks like an empty root span.

A similar issue looks to have be recently fixed in isolated function see here: https://github.com/Azure/azure-functions-dotnet-worker/issues/2733
 

#### Repro steps

1. Create a new HTTP Trigger function (in-proc) running on net6

2. Add OpenTelemetry configuration

```
  builder.Services.AddOpenTelemetry().WithTracing(tracerProviderBuilder =>
          {
              tracerProviderBuilder
                  .SetResourceBuilder(appResourceBuilder)
                  .SetSampler<AlwaysOnSampler>()
                  .AddSource("ServiceName")
                  .AddOtlpExporter();
          });
```
3. Add a breakpoint after function dispatch, add watch on `Activty.Current` => should be null

4. Repeat above steps with net8. Should see something similar to the below

![Image](https://github.com/user-attachments/assets/fa882056-9635-4512-9a08-f4e3c9414ca0)

#### Expected Behaviour
Expected behaviour is not having the host create a current activity. If host is required to create activities then a source name is required to be able to subscribe to it. This results in missing spans in our otel telemetry after export.

#### Known workarounds
We've temporarily patched this by setting `Activty.Current` to null before starting a root activity.

**Comment by bhagyshricompany** [comment]
Hi @JessicaA-Auror  thanks for update this is occurring exactly on which net 6 or net 6. will  check and update.Thanks

**Comment by jviau** [comment]
@JessicaA-Auror I don't think this is functions specific. I believe this is a BCL change between .net6 and .net8. You are specifically seeing an operation from AspNetCore. Have you tried using OTels AspNetCore instrumentation package? I believe they do some work to shim this: https://www.nuget.org/packages/OpenTelemetry.Instrumentation.AspNetCore

**Comment by RohitRanjanMS** [comment]
I will go ahead and remove the empty activity.
